### ISSUE TITLE
Accessible inline marks for HTML5

### ISSUE THREAD

**Issue by dginev** [brucemiller/LaTeXML#2066]
I was testing some extensions providing accessible readouts and all of them seem to treat `<cite>` akin to `<span>`, i.e. without adding any custom treatment. That lead me to re-reading some resources mentioning `<cite>` and they kept emphasizing "a title of a work".

The [spec text](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-cite-element) also emphasizes not using the element for non-title content by saying:
> The [cite](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-cite-element) element [represents](https://html.spec.whatwg.org/multipage/dom.html#represents) the title of a work. [...] A person's name is not the title of a work — even if people call that person a piece of work — and the element must therefore not be used to mark up people's names.

It appears that accessibility tools have taken that to heart, and treat the `<cite>` as transparent, assuming the title can self-vocalize just fine in context.

I also proof-listened to some inline citations in HTML pages of known journals. None of them vocalized well, and most used simple `<a>` elements, to link to the backmatter bibliography. Some used `<sup>` to stylize the inline cites as note marks. Through a speech engine, the content sounds a bit like playing bingo, e.g.
> ... previous work 13,17,21 has shown...

There is a related report from [ar5iv#95](https://github.com/dginev/ar5iv/issues/95), which specifically found that Mac VoiceOver can not interpret the `<cite>` markup emitted by latexml. That reporter mentioned that having individual links, or one link per cite element, seems to succeed in that AT.

In conclusion, I am tempted to suggest moving away from using `<cite>` for inline citations. Or alternatively, starting a conversation with the HTML5 team about their stance on using `<cite>` for inline citation labels. And in either case we should consider enhancing the markup so that accessibility tools can clearly announce the number is a citation (and relatedly, when a number is a footnote mark, and when we emit `<li>` ordered list items which have a custom marker - currently they get vocalized with both a number and a marker).

**Comment by dginev** [comment]
After some digging, I stumbled on one publisher which is already emitting dedicated inline citation markup for accessibility, namely the AMS. Here is an [example article](https://www.ams.org/journals/notices/202305/noti2690/noti2690.html).

They have leveraged a DAISY specification called [DPUB-ARIA](http://kb.daisy.org/publishing/docs/html/dpub-aria/), which introduces a role called [doc-biblioref](http://kb.daisy.org/publishing/docs/html/dpub-aria/doc-biblioref.html). That leads to the HTML markup:
```html
<cite><a role="doc-biblioref" data-ams-ref="bibr" href="#bibr-thurston">Thu88</a></cite>
```

But putting a pragmatic hat on again, I have no answers whether there is any HTML/ePub "screen reader" application that is already consuming this specification. My [read-aloud](https://addons.mozilla.org/en-US/firefox/addon/read-aloud/) debugging clearly doesn't - I think that tool just barely uses some of the standard ARIA attributes. So it will not be a straightforward choice akin to "everyone is already consuming DPUB-ARIA, we should emit it". Though it may be a good idea to consider emitting it irrespective.

Side-notes: 
  1. a curious choice the AMS HTML has made is that the bracketed `[` `]` enclosure for `[Thu88]` is provided via CSS `::before` and `::after` rules, and the characters are missing from the HTML content.
  2. also curious is that the `<cite>` element is indeed used as an outer wrapper, even if it doesn't seem to have any application-specific effect (and similarly to latexml's use, doesn't agree with the HTML spec of `<cite>`). The element isn't even used to anchor CSS rules, from what I can tell (again a similarity with latexml, where we rely on a `.ltx_cite` class selector instead).

**Comment by dginev** [comment]
A very nice additional resource I can attach today is an accessible walkthrough for on of the ar5iv article, as converted with latexml, as prepared by Patrick Smyth from the Iota School:
https://youtu.be/bHM6mes51nE

Performed via NVDA, Windows 10, Chrome. 

It raises some nice questions, and more importantly - demonstrates what is generated by a current accessibility tool that is encountering our markup.

**Comment by gjw32** [comment]
I have not encountered any screen readers that do anything special with the <cite> element.
Regarding footnotes, there are two approaches that I can think of. The first is the typical list of footnotes at the end of the webpage with a link from the footnote reference to the footnote content and a link back to the reference (see [https://www.sitepoint.com/accessible-footnotes-css/](https://www.sitepoint.com/accessible-footnotes-css/) for an implementation.) The second option would be to have the reference as a button which opens a modal dialog with the footnote content using the <dialog> HTML element.

**Comment by dginev** [comment]
Some context to the last comment by @gjw32 : we had a brief exchange following a latexml mailing list question, where he made the excellent point that LaTeXML's current treatment of \footnote will be undetectable by screen readers. To which I replied that we should likely consider the available ARIA options and enhance our HTML markup.

One link to offer here describes the `role="note"` approach:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/roles/note_role

We would likely need to combine that with an `aria-labelledby` id pointer, to avoid always reading all the decorative footnote marks, but that is best explored while testing with a compliant screen reader or two.

**Comment by brucemiller** [comment]
There seems to be several things going on here, conflated with "inline marks".

The spec on `cite` seems poorly thought out, or at best, poorly explained. The choice of the name "cite" would seem to suggest that it has *something* to do with citation. OTOH the claim that it should only contain a "title" and apparently even asserting that it should be used *within* a bibliographic item for the item's title, suggests that it is like `<title>` but for something else's title.
Then there's the point that, in a LaTeX-y sense, what is contained within a `\cite{...}` exactly *is* the *local* title for the given resource.

Moreover, it isn't even clear to me that AT should do anything special with `cite` given *either* interpretation, or what it should do.  The several visual styles (parenthesized, bracketed, superscript) are intended to separate them from the running text, whereas unbracketed author-year style intends them to be read as if they were part of the text. But then, both of those cases are already largely preserved in the output. It is no wonder that AT hasn't done anything.  That AT doesn't treat `cite` specially seems hardly a reason to replace it by `span` which AT doesn't treat specially. There's certainly an open question on whether we should provide any additional markup, aria or otherwise, but seems to me like the AT community doesn't yet know what it should be?

There was a potentially valid point made  that *perhaps* we shouldn't combine multiple citations into a single `cite`? 

I'm confused about "footnote marker", whether you're talking about footnotes, or still about citations. I can imagine that proper footnotes could need some help, since there isn't any specific html markup annotation used, and they way they appear in the html (popup, margin, etc) can vary.

All of this seems speculative for the upcoming release, however.

**Comment by dginev** [comment]
> All of this seems speculative for the upcoming release, however.

Agreed, I mostly wanted to have an early review to see if we had any easy wins for the Fall accessibility testing efforts based around latexml HTML.

Since there are various open questions without easy answers, might as well push back a milestone.
